### PR TITLE
Autocomplete with label

### PR DESCRIPTION
**Pull request by alonshmiel1**
- [ ] - Storybook description tab header + examples should contain options
- [ ] - Clicking on the chevron should trigger input focus
- [ ] - Chevron should be vertically centered
- [ ] - Add testkit documentation
- [ ] - Add e2e tests
- [ ] - Add tests for `InputWithOptions` predicate prop + documentation
- [ ] - Verify that onSelect is working in uncontrolled mode

### CODE REVIEW

File: src/InputWithOptions/InputWithOptions.js [hunk 1]
```diff
@@ -109,9 +109,12 @@ class InputWithOptions extends Component {
   }
```
Comment (nitayneeman) [inline src/InputWithOptions/InputWithOptions.js]: What is the motivation for these changes?

Let's do them in a separate PR and also make sure to cover them by appropriate tests

File: src/AutocompleteWithLabel/AutocompleteWithLabel.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      this.setState({ value: event.target.value });
+    };
+
+    const {
+      label,
+      options,
+      value,
+      dataHook,
+      size: _size,
+      ...rest
```
Comment (nitayneeman) [inline src/AutocompleteWithLabel/AutocompleteWithLabel.js]: Let's remove that, so we'll use only the necessary props (once we need some <Input/> props, we would add them explicitly)

**Comment by nitayneeman** [comment]
@yakirn Do you know who is going to complete this PR?

**Comment by yakirn** [comment]
@nitayneeman best consult with @assafswix

**Comment by nitayneeman** [comment]
> @nitayneeman best consult with @assafswix

@assafswix  I'm closing this PR at the moment.
Let's reopen when there is someone who's dedicated to complete the work.

File: src/AutocompleteWithLabel/docs/index.story.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  exampleProps: {
+    options: [{ label: 'options', value: options }],
+  },
+
+  sections: [
+    header({
+      issueUrl: 'https://github.com/wix/wix-style-react/issues/new',
+      sourceUrl:
+        'https://github.com/wix/wix-style-react/tree/master/src/AutocompleteWithLabel/',
+      component: <AutocompleteWithLabel label="label" />,
```
[the remark this answers is not in the document]
Comment (saarkuriel) [reply src/AutocompleteWithLabel/docs/index.story.js]: What part?

[the remark this answers is not in the document]
Comment (nitayneeman) [reply src/AutocompleteWithLabel/docs/index.story.js]: `component: <AutocompleteWithLabel label="label" />`

Another option is adding there a better example instead, but we'll have anyway as part of the Story examples below, so I wouldn't invest effort on that

File: .wuf/testkits/definitions.js [hunk 4]
```diff
@@ -727,10 +721,7 @@ module.exports = {
   },
 
   AutoCompleteComposite: {
-    vanillaLegacyTestkit: true,
-    enzymeLegacyTestkit: true,
-    protractorLegacyTestkit: true,
-    noPuppeteerTestkit: true,
+    noPuppeteerTestkit: true, // was not exported before automation, keeping that way for now
```
Comment (nitayneeman) [inline .wuf/testkits/definitions.js]: This fails the CI. 

It should be:
```suggestion
    vanillaLegacyTestkit: true,
    enzymeLegacyTestkit: true,
    protractorLegacyTestkit: true,
    noPuppeteerTestkit: true,
```

File: .wuf/components.json [hunk 5]
```diff
@@ -400,5 +400,8 @@
   },
   "Divider": {
     "path": "src/Divider"
+  },
+  "LabelledElement": {
```
Comment (argshook) [inline .wuf/components.json]: Where are all these changes from? Rebasing on latest master should not require any of the generated changes

**Comment by argshook** [comment]
note for whoever is going to work next on this:

i'd prefer component to be:
```diff
-AutocompleteWithLabel
+AutoCompleteWithLabel
```

that's simply because we already have `AutoComplete` and `AutoCompleteComposite`.

yes, it's quite a requirement for a very minor thing but we already have
many tiny inconcistencies like that and when they add up, it causes confusion.

if whoever is going to work next on this has troubles with mass rename,
please contact me, i'm happy to help

File: src/AutoCompleteWithLabel/test/AutoCompleteWithLabel.visual.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    storiesOf(`AutoCompleteWithLabel/${describe}`, module).add(it, () => (
+      <AutoCompleteWithLabel {...commonProps} {...props} />
+    ));
+  });
+});
+
+interactiveTests.forEach(({ describe, its }) => {
+  its.forEach(({ it, props, componentDidMount }) => {
+    storiesOf(`InputWithLabel/${describe}`, module).add(it, () => (
+      <InteractiveEyeTest {...props} componentDidMount={componentDidMount} />
```
Comment (nitayneeman) [inline src/AutoCompleteWithLabel/test/AutoCompleteWithLabel.visual.js]: Minor but we could use there `commonProps` as well:
 ```suggestion
      <InteractiveEyeTest {...commonProps} {...props} componentDidMount={componentDidMount} />
```

File: src/AutoCompleteWithLabel/docs/index.story.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  component: AutoCompleteWithLabel,
+  componentPath: '..',
+
+  componentProps: {
+    value: '',
+    label: 'my label',
+  },
+
+  exampleProps: {
+    options: [{ label: 'options', value: options }],
```
Comment (nitayneeman) [inline src/AutoCompleteWithLabel/docs/index.story.js]: Do you have some good example of `suffix` to add there?

Currently when you try to insert some text within the "suffix" field in Storybook it throws an error that makes the page be blank

File: src/AutoCompleteWithLabel/AutoCompleteWithLabel.js [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  static propTypes = {
+    dataHook: PropTypes.string,
+    /** label to apear in input */
+    label: PropTypes.string.isRequired,
+    options: PropTypes.arrayOf(optionValidator).isRequired,
+    suffix: PropTypes.arrayOf(PropTypes.element),
+    status: PropTypes.oneOf([
+      Input.StatusError,
+      Input.StatusWarning,
+      Input.StatusLoading,
```
Comment (nitayneeman) [inline src/AutoCompleteWithLabel/AutoCompleteWithLabel.js]: Maybe it's worth to write the values explicitly hardcoded because our AutoDocs doesn't know to deal with that at the moment

File: src/AutoCompleteWithLabel/docs/index.story.js (hunk 7 above)
Comment (saarkuriel) [reply src/AutoCompleteWithLabel/docs/index.story.js]: I think that since `InputWithLabel` provides this options, this component should too cause this is the same UX...

**Comment by saarkuriel** [comment]
> 3. Hover cursor should be pointer
>    image

This is the same as in `InputWithOptions`, cause once you click on the component you can start writing. Are you sure I should change it?

**Comment by LittleTal** [comment]
@saarkuriel Ok, no problem, let's keep it as it is
